app.rb

```diff
@@ -74,32 +74,30 @@ def handle_event(event:, context:)
 def get_holding(event)
   $logger.info('handling get request')
   params = event['queryStringParameters']
-  $logger.info('params: ', params)
-  if ids = params['id']
-    $logger.info("getting by ids: #{ids}")
-    begin
-      parsed_ids = ids.split(",").map {|id| id.to_i}
-      records = Record.find(parsed_ids)
-      return respond(200, records.map {|record| record.to_json})
-    rescue => e
-      $logger.error("problem getting records with ids: #{ids}", e.message)
-      return respond(500, "problem getting records with ids: #{ids}, message: #{e.message}")
-    end
-  elsif bib_ids = params['bib_ids']
-    offset = params['offset'] ? params['offset'].to_i : 0
-    limit = params['limit'] ? params['limit'].to_i : 20
-    $logger.info("getting by bib_ids: #{bib_ids}, offset: #{offset}, limit: #{limit}")
-    begin
-      parsed_bib_ids = bib_ids.split(",").map {|id| id.to_i}
-      records = Record.where("bib_ids && ARRAY[?]::int[]", parsed_bib_ids).offset(offset).limit(limit)
-      return respond(200, records.map {|record| record.to_json})
-    rescue => e
-      $logger.error("problem getting records with bib_ids: #{bib_ids}, offset: #{offset}, limit: #{limit}", e.message)
-      return respond(500, "problem getting records with bib_ids: #{bib_ids}, offset: #{offset}, limit: #{limit} message: #{e.message}")
-    end
+  $logger.info("params: #{params}")
+  if params && ids = params['ids']
+    getting_by = 'ids'
+    identifier_for_where = 'ARRAY[id]::int[]'
+  elsif params && ids = params['bib_ids']
+    getting_by = 'bib_ids'
+    identifier_for_where = 'bib_ids'
   else
-    $logger.info("Missing required fields ids or bib_ids")
-    return respond(400, "Missing required fields ids or bib_ids")
+    message = "Missing required fields ids or bib_ids"
+    $logger.info(message)
+    return respond(400, message)
+  end
+  offset = params['offset'] ? params['offset'].to_i : 0
+  limit = params['limit'] ? params['limit'].to_i : 20
```

Comment:
Since this only responds to queries with specific record identifiers, I'm not sure the limit/offset parameters are necessary? Bibs can have multiple holdings but I can't think of a situation where you would want to page through them.

I can see the argument for including it, but I think it suggests a use case that we aren't really supporting